lib/retry.go

```diff
@@ -0,0 +1,25 @@
+package lib
+
+import (
+	"context"
+	"errors"
+	"time"
+)
+
+type RetryOptions struct {
+	RetriesCount    int
+	RetriesInterval time.Duration
+}
+
+func WithRetries(ctx context.Context, retryFn func() error, options RetryOptions) error {
+	for i := 0; i < options.RetriesCount; i++ {
+		if err := ctx.Err(); err != nil {
+			return err
+		}
```

Comment:
O correto seria Colocar um select pegando o ``Done()` do contexto
Ficaria algo assim

```
		select {
		case <-ctx.Done():
			if err := ctx.Err(); err != nil {
				return err

			}
			return nil
		default:
			resto da func
		}

Reply:
@lffspaniol qual seria o problema em usar `ctx.Err()` diretamente nesse caso? Parece ser uma forma equivalente, apenas com a diferença de usar _channel_. Certo? 🤔

Reply:
Vc só vai verificar nil sempre e deixar passar

Reply:
Quando eu rodo o teste ele consegue capturar o erro mesmo sem o select

Reply:
Pelo que entendi na doc, ou é usado o select para capturar o Done() ou é usada a ferificação do contexto cancelado diretamente com o Err().

Reply:
// Se Done ainda não estiver fechado, Err retorna nil. 
// Se Done for fechado, Err retornará um erro não nulo explicando o motivo: 
// Cancelado se o contexto foi cancelado 
// ou DeadlineExceeded se o prazo final do contexto tiver passado. 
// Após Err retornar um erro não nulo, chamadas sucessivas para Err retornam o mesmo erro. 

[ctx.Err vs ctx.Done](https://stackoverflow.com/questions/67439974/ctx-err-vs-ctx-done-chan-closed-check-is-there-a-difference#:~:text=If%20you're%20checking%20if,for%20checking%20for%20an%20error.)

[Context Package](https://pkg.go.dev/context#Context)

[Go Concurrency Patterns](https://go.dev/blog/pipelines)